commit 76b05c939a925e96a227106af08971357e36d579
Author: Carolyn MacLeod <carolyn>
Date:   Fri Dec 3 23:17:03 2004 +0000

    temporary fix for bug 79391

1	0	examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/SliderTab.java
diff --git a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/SliderTab.java b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/SliderTab.java
index 00be184..cf8ecef 100755
--- a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/SliderTab.java
+++ b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/SliderTab.java
@@ -191,6 +191,7 @@ class SliderTab extends RangeTab {
 	 * Gets the text for the tab folder item.
 	 */
 	String getTabText () {
+		if (org.eclipse.swt.internal.Platform.PLATFORM.equals("carbon")) return "S/S";
 		return "Slider/Scale";
 	}
 	
commit d748516422f8e11519375f9bfd8434c434a261b2
Author: Carolyn MacLeod <carolyn>
Date:   Thu Mar 2 21:16:15 2006 +0000

    bug 79391

14	5	examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/ControlExample.java
7	0	examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/CoolBarTab.java
1	1	examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/CustomControlExample.java
7	0	examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/ExpandBarTab.java
7	0	examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/ProgressBarTab.java
7	1	examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/SliderTab.java
9	0	examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/Tab.java
7	0	examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/TabFolderTab.java
7	0	examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/ToolBarTab.java
diff --git a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/ControlExample.java b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/ControlExample.java
index c196a74..738974b 100755
--- a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/ControlExample.java
+++ b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/ControlExample.java
@@ -25,6 +25,7 @@ public class ControlExample {
 		ResourceBundle.getBundle("examples_control"); //$NON-NLS-1$
 	private ShellTab shellTab;
 	private TabFolder tabFolder;
+	private Tab [] tabs;
 	Image images[];
 
 	static final int ciClosedFolder = 0, ciOpenFolder = 1, ciTarget = 2, ciBackground = 3, ciParentBackground = 4;
@@ -52,7 +53,7 @@ public class ControlExample {
 	public ControlExample(Composite parent) {
 		initResources();
 		tabFolder = new TabFolder (parent, SWT.NONE);
-		Tab [] tabs = createTabs();
+		tabs = createTabs();
 		for (int i=0; i<tabs.length; i++) {
 			TabItem item = new TabItem (tabFolder, SWT.NONE);
 		    item.setText (tabs [i].getTabText ());
@@ -199,7 +200,7 @@ public class ControlExample {
 		shell.setLayout(new FillLayout());
 		ControlExample instance = new ControlExample(shell);
 		shell.setText(getResourceString("window.title"));
-		setShellSize(shell);
+		setShellSize(instance, shell);
 		shell.open();
 		while (! shell.isDisposed()) {
 			if (! display.readAndDispatch()) display.sleep();
@@ -217,12 +218,20 @@ public class ControlExample {
 	/**
 	 * Sets the size of the shell to it's "packed" size,
 	 * unless that makes it larger than the monitor it is being displayed on,
-	 * in which case set it to slightly smaller than the monitor.
+	 * in which case we may be able to recover some width by using shorter tab names.
+	 * If that fails, just set the shell size to be slightly smaller than the monitor.
 	 */
-	static void setShellSize(Shell shell) {
+	static void setShellSize(ControlExample instance, Shell shell) {
 		Point size = shell.computeSize(SWT.DEFAULT, SWT.DEFAULT);
 		Rectangle monitorArea = shell.getMonitor().getClientArea();
-		shell.setSize(Math.min(size.x, monitorArea.width - 20), Math.min(size.y, monitorArea.height - 20));
+		if (size.x > monitorArea.width) {
+			TabItem [] tabItems = instance.tabFolder.getItems();
+			for (int i=0; i<tabItems.length; i++) {
+				tabItems[i].setText (instance.tabs [i].getShortTabText ());
+			}
+			size = shell.computeSize(SWT.DEFAULT, SWT.DEFAULT);
+		}
+		shell.setSize(Math.min(size.x, monitorArea.width), Math.min(size.y, monitorArea.height));
 	}
 }
 
diff --git a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/CoolBarTab.java b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/CoolBarTab.java
index d0a5b54..957f909 100644
--- a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/CoolBarTab.java
+++ b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/CoolBarTab.java
@@ -282,6 +282,13 @@ class CoolBarTab extends Tab {
 	}
 
 	/**
+	 * Gets the short text for the tab folder item.
+	 */
+	public String getShortTabText() {
+		return "CB";
+	}
+
+	/**
 	 * Gets the text for the tab folder item.
 	 */
 	String getTabText () {
diff --git a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/CustomControlExample.java b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/CustomControlExample.java
index cb4cfa2..7c8998f 100644
--- a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/CustomControlExample.java
+++ b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/CustomControlExample.java
@@ -48,7 +48,7 @@ public class CustomControlExample extends ControlExample {
 		shell.setLayout(new FillLayout());
 		CustomControlExample instance = new CustomControlExample(shell);
 		shell.setText(getResourceString("custom.window.title"));
-		setShellSize(shell);
+		setShellSize(instance, shell);
 		shell.open();
 		while (! shell.isDisposed()) {
 			if (! display.readAndDispatch()) display.sleep();
diff --git a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/ExpandBarTab.java b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/ExpandBarTab.java
index 3a4f130..6109885 100644
--- a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/ExpandBarTab.java
+++ b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/ExpandBarTab.java
@@ -119,6 +119,13 @@ class ExpandBarTab extends Tab {
 	}
 	
 	/**
+	 * Gets the short text for the tab folder item.
+	 */
+	public String getShortTabText() {
+		return "EB";
+	}
+
+	/**
 	 * Gets the text for the tab folder item.
 	 */
 	String getTabText () {
diff --git a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/ProgressBarTab.java b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/ProgressBarTab.java
index cc0b59d..f14b5fa 100755
--- a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/ProgressBarTab.java
+++ b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/ProgressBarTab.java
@@ -92,6 +92,13 @@ class ProgressBarTab extends RangeTab {
 	}
 
 	/**
+	 * Gets the short text for the tab folder item.
+	 */
+	public String getShortTabText() {
+		return "PB";
+	}
+
+	/**
 	 * Gets the text for the tab folder item.
 	 */
 	String getTabText () {
diff --git a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/SliderTab.java b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/SliderTab.java
index 26905c3..d26efb1 100755
--- a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/SliderTab.java
+++ b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/SliderTab.java
@@ -187,10 +187,16 @@ class SliderTab extends RangeTab {
 	}
 
 	/**
+	 * Gets the short text for the tab folder item.
+	 */
+	public String getShortTabText() {
+		return "S/S";
+	}
+
+	/**
 	 * Gets the text for the tab folder item.
 	 */
 	String getTabText () {
-		if (SWT.getPlatform().equals("carbon")) return "S/S";
 		return "Slider/Scale";
 	}
 	
diff --git a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/Tab.java b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/Tab.java
index a5ad2fb..c07f0f7 100755
--- a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/Tab.java
+++ b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/Tab.java
@@ -1001,6 +1001,15 @@ abstract class Tab {
 	}
 	
 	/**
+	 * Gets the short text for the tab folder item.
+	 *
+	 * @return the short text for the tab item
+	 */
+	public String getShortTabText() {
+		return getTabText();
+	}
+
+	/**
 	 * Gets the text for the tab folder item.
 	 *
 	 * @return the text for the tab item
diff --git a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/TabFolderTab.java b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/TabFolderTab.java
index d0cb799..3cf542e 100644
--- a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/TabFolderTab.java
+++ b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/TabFolderTab.java
@@ -152,6 +152,13 @@ class TabFolderTab extends Tab {
 	}
 
 	/**
+	 * Gets the short text for the tab folder item.
+	 */
+	public String getShortTabText() {
+		return "TF";
+	}
+
+	/**
 	 * Gets the text for the tab folder item.
 	 */
 	String getTabText () {
diff --git a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/ToolBarTab.java b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/ToolBarTab.java
index ca4629a..e94412b 100755
--- a/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/ToolBarTab.java
+++ b/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/ToolBarTab.java
@@ -281,6 +281,13 @@ class ToolBarTab extends Tab {
 	}
 	
 	/**
+	 * Gets the short text for the tab folder item.
+	 */
+	public String getShortTabText() {
+		return "TB";
+	}
+
+	/**
 	 * Gets the text for the tab folder item.
 	 */
 	String getTabText () {
